packages/server/src/tests/header/headerSelection.ts

```diff
@@ -107,91 +91,96 @@ const supportersTest: HeaderTest = {
     ],
 };
 
-const supportersTestUS: HeaderTest = {
-    name: 'header-supporter__USEOY',
-    audience: 'AllExistingSupporters',
-    variants: [
-        {
-            name: 'control',
-            modulePathBuilder: usEOYHeader.endpointPathBuilder,
-            content: {
-                heading: 'Thank you for supporting us',
-                subheading: '',
-                primaryCta: {
-                    url:
-                        'https://support.theguardian.com/us/contribute?selected-contribution-type=ONE_OFF',
-                    text: 'Make an extra contribution',
-                },
-            },
-            mobileContent: {
-                heading: 'Thank you',
-                subheading: '',
-                primaryCta: {
-                    url:
-                        'https://support.theguardian.com/us/contribute?selected-contribution-type=ONE_OFF',
-                    text: 'Contribute again',
-                },
-            },
-        },
-    ],
-};
-
-const usEoyPeriodStart = new Date(2021, 10, 22);
-const usEoyPeriodEnd = new Date(2022, 1, 1);
+const existingSupportersAndEveryone = ['Everyone', 'AllExistingSupporters'];
 
-const isInUsEoyPeriod = (date: Date): boolean => {
-    return isAfter(date, usEoyPeriodStart) && isBefore(date, usEoyPeriodEnd);
-};
+const selectBestTest = (
+    targeting: HeaderTargeting,
+    allTests: HeaderTest[],
+): HeaderTestSelection => {
+    // If there is a need to return other specific hardcoded tests/variants, include required logic to identify, and required return, here
 
-const hasNotContributedDuringUsEoyPeriod = (lastOneOffContributionDate?: string): boolean => {
-    if (!lastOneOffContributionDate) {
-        return true;
+    // In case the allTests array is empty, return a hardcoded test/variant
+    if (!allTests.length) {
+        return selectHardcodedTest(targeting);
     }
 
-    return isBefore(new Date(Date.parse(lastOneOffContributionDate)), usEoyPeriodStart);
-};
+    // Cascade through allTests to match supplied targeting data to the first test that meets requirements - if no test meets requirements, return an appropriate hardcoded test/variant
+    let selectedTest;
+
+    const { showSupportMessaging, countryCode } = targeting;
+
+    const countryGroupId = countryCodeToCountryGroupId(countryCode.toUpperCase());
+
+    for (let i = 0, iz = allTests.length; i < iz; i++) {
+        const test = allTests[i];
+
+        const { isOn, userCohort, locations } = test;
 
-const getNonSupportersTest = (edition: Edition): HeaderTest => {
-    if (edition === 'UK') {
-        return nonSupportersTestUK;
+        if (isOn) {
+            if (showSupportMessaging && existingSupportersAndEveryone.indexOf(userCohort) >= 0) {
```

Comment:
what about `AllNonSupporters`?

Reply:
Question: what is the purpose of the showSupportMessaging flag? How is it supposed to interact with the user's userCohort value? The UserCohort type (should it be an enum?) is currently set in the shared.ts file to:

`export type UserCohort = 'AllExistingSupporters' | 'AllNonSupporters' | 'Everyone' | 'PostAskPauseSingleContributors';`

Reply:
It's a really unfortunate name, chosen long ago. Essentially `showSupportMessaging` means "user is not a supporter". The client sends `true` if the user is not a supporter.
I think it's best to do what the banner does: https://github.com/guardian/support-dotcom-components/blob/11353321bb2a961d7fe4f71e6e62a899ec26d877/packages/server/src/tests/banners/bannerSelection.ts#L76

`PostAskPauseSingleContributors` is not actually supported by the RRCP and should be removed, but don't worry about this right now.